.github/comment.yml

```diff
@@ -3,11 +3,37 @@
     type: checkbox
     keywords:
       - "WaitForARMFeedback"
-    booleanFilterExpression: "!(ARMSignedOff||ARMChangesRequested||Approved-OkToMerge||WaitForARMRevisit||BreakingChangeReviewRequired)"
+    booleanFilterExpression: "!(ARMSignedOff||ARMChangesRequested||Approved-OkToMerge||WaitForARMRevisit||BreakingChangeReviewRequired||NewApiVersionRequired)"
```

Comment:
When `NewApiVersionRequired` label gets added too late, this rule will fire instead of the rule added below, potentially leading to confusion, as these rules result in different comment and guidance. The labels should eventually be corrected by the rule that reacts to addition to the `NewApiVersionRequired`.

I think we should deploy these changes and observe how often these rules fire out of order, and how much problems it is causing. If significant amount, then we can work on ensuring  proper ordering. This likely will be quite nontrivial.

Reply:
Is it worth having both of these rules, as opposed to just adding a note about this in the comment?

Also I'm wondering for these checkbox workflows do they take effect when someone checks the box as they open the PR? What about if they check or uncheck it after the PR is opened. Will we end up with potentially a lot of these comments?

Reply:
@weshaggard 

Re: 

> Is it worth having both of these rules, as opposed to just adding a note about this in the comment?

These rules result in different sets of labels. The first rule produces `WaitForARMFeedback` and `ARMReview` while the 2nd rule produces `ARMChangesRequested` and `ARMReview`.

These labels have specific rules about who (automation vs human) should add them, per [the diagram](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/909/API-spec-PR-review?anchor=management-plane-api-review) and [issue #6076](https://github.com/Azure/azure-sdk-tools/issues/6076):

![image](https://github.com/Azure/azure-rest-api-specs/assets/4429827/31be00cd-5ec0-4726-aad0-95c46117dbe0)

Knowing that, how would you suggest to change this? 

Re: 

> Also I'm wondering for these checkbox workflows do they take effect when someone checks the box as they open the PR? What about if they check or uncheck it after the PR is opened. Will we end up with potentially a lot of these comments?

My read of the source is that the rule triggers:

- upon PR edit, as can be seen [here](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/openapi-alps?path=private/openapi-kebab/src/bots/workflow/workflowBot.ts&version=GC9df1258de7cdcbd322cfc2ff15f4ef4f96998f9d&line=870&lineStartColumn=29&lineEndColumn=57&_a=contents
) (it is in `handlePullRequestEdited`)
- and also upon the PR creation, because the PR edit handler is called from `handlePullRequestOpened`, as can be seen [here](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/openapi-alps?path=private/openapi-kebab/src/bots/workflow/workflowBot.ts&version=GC9df1258de7cdcbd322cfc2ff15f4ef4f96998f9d&line=581&lineStartColumn=16&lineEndColumn=39&_a=contents).

I think it won't duplicate the comment, because it checks for its existence, [here](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/openapi-alps?path=private/openapi-kebab/src/bots/workflow/workflowBot.ts&version=GC9df1258de7cdcbd322cfc2ff15f4ef4f96998f9d&line=939&lineEnd=942&lineStartColumn=1&lineEndColumn=6&_a=contents)

Reply:
>These rules result in different sets of labels. The first rule produces WaitForARMFeedback and ARMReview while the 2nd rule produces ARMChangesRequested and ARMReview.

Perhaps I'm missing something but I think these rules only come into play when a checkbox is checked in the PR and they use these label filters:
```
"!(ARMSignedOff||ARMChangesRequested||Approved-OkToMerge||WaitForARMRevisit||BreakingChangeReviewRequired||NewApiVersionRequired)"
"!(ARMSignedOff||ARMChangesRequested||Approved-OkToMerge||WaitForARMRevisit||BreakingChangeReviewRequired)&&NewApiVersionRequired"
```

With the only difference being the `NewApiVersionRequired` label. However I don't think the `NewApiVersionRequired` label will ever be there on creation of a PR and so it will only ever exist if someone checks to box for ARM review required in after the PR is opened and after the checks have ran. Am I not understanding how this rule gets triggered?

Reply:
I think what you described is the most common scenario, but I think somebody may come in later and check this checkbox. E.g. the PR assignee looks at the PR and realizes the PR author forgot to add the checkbox, and checks it themselves. This action of checking the checkbox happens some time after the PR was opened, and also after the `NewApiVersionRequired` label was added.

Reply:
@weshaggard on a second thought, I think we should keep both rules. The 2nd rule can definitely happen per the scenario I described above. If we remove it, folks would have to instead remember to manually add the `ARMReview` label, without which the automation won't work correctly (see usages of this label in the `comment.yml`.